src/util.c

```diff
@@ -162,20 +168,27 @@ struct passwd *hyper_getpwnam(const char *name)
 		struct passwd *pwd = fgetpwent(file);
 		if (!pwd)
 			break;
-		if (!strcmp(pwd->pw_name, name) || pwd->pw_uid == uid) {
```

Comment:
just 
`if ((INVALID_UGID == uid && !strcmp(pwd->pw_name, name)) || pwd->pw_uid == uid) {`
is enough.
INVALID_UGID should be changed to a better name.

Reply:
The test case is above this pr, take hyper_getpwnam(1000) for example
```
/etc/passwd                              
1000:x:1001:0:*:/*:/bin/bash
                   |
                   V
hyper_getpwnam(1000)  = struct passwd {pw_name: 1000, uid: 1001} 
```
```
/etc/passwd                              
1000:x:1001:0:*:/*:/bin/bash
hello:x:1000:0:*:/*:/bin/bash
                   |
                   V
hyper_getpwnam(1000)  = struct passwd {pw_name: hello, uid: 1000} 
```
if here use "if ((INVALID_UGID == uid && !strcmp(pwd->pw_name, name)) || pwd->pw_uid == uid) "

```
/etc/passwd                              
1000:x:1001:0:*:/*:/bin/bash
hello:x:1000:0:*:/*:/bin/bash
                   |
                   V
hyper_getpwnam(1000)  = struct passwd {pw_name: 1000, uid: 1001} 
```
```
/etc/passwd                              
have no 1000 as name or uid
                   |
                   V
hyper_getpwnam(1000)  = NULL
```

And what do you think we can use to replace INVALID_UGID ?